pkg/lib/operationclient.go

```diff
@@ -150,6 +156,46 @@ func (oc *OperationClient) BdevGetIosat(bdevName string) (*Iostat, error) {
 	}, nil
 }
 
+func (oc *OperationClient) BdevGetHistogram(bdevName string) (*Histogram, error) {
+	logger.Info("BdevGetHistogram: bdevName %v", bdevName)
+	params := &struct {
+		Name string `json:"name"`
+	}{
+		Name: bdevName,
+	}
+
+	rsp := &struct {
+		Error  *spdkErr `json:"error"`
+		Result *struct {
+			Histogram   string
+			TscRate     uint64
+			BucketShift uint32
+		}
+	}{}
+
+	err := oc.sc.Invoke("bdev_get_histogram", params, rsp)
+	if err != nil {
+		logger.Error("bdev_get_histogram failed: %v", err)
+		return nil, err
+	}
+
+	if rsp.Error != nil {
+		logger.Error("bdev_get_histogram rsp err: %v", *rsp.Error)
+		return nil, fmt.Errorf("bdev_get_histogram rsp err: %d %s", rsp.Error.Code, rsp.Error.Message)
+	}
+
+	if rsp.Result == nil {
+		logger.Error("bdev_get_histogram result is nil")
```

Comment:
A pd might be deleted after we get the pd name list and before we call bdev_get_histogram. So we may get a nil result or an error like "the device is not found". We should set log level to Warn in that case. For simplicity, we may set the log level to Warn for all rsp.Error != nil or rsp.Result == nil.

Reply:
good to know